src/main/java/seedu/address/model/person/PersonUtil.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.person;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+
+/**
+ * Utility class for handling role-specific information for different types of Person instances.
+ */
+public class PersonUtil {
+
+
+    /**
+     * Retrieves role-specific information based on the role of the given {@code Person}.
+     * <p>
+     * This method determines the role of the {@code Person} instance and extracts
+     * specific information related to that role:
+     * <ul>
+     *     <li>If the person is a {@code Volunteer}, it retrieves the volunteer's hours as a string.</li>
+     *     <li>If the person is a {@code Donor}, it retrieves the donor's donated amount as a string.</li>
+     *     <li>If the person is a {@code Partner}, it retrieves the partner's partnership end date as a string.</li>
+     * </ul>
+     * </p>
+     *
+     * @param person the {@code Person} instance whose role-specific information is to be retrieved.
+     * @return a {@code String} containing the role-specific information.
+     */
+    public static String getRoleSpecificInfoString(Person person) throws CommandException {
```

Comment:
Should this be changed? Since unless im missing something, there are no `CommandException` that will be thrown